### PR TITLE
Deprecate the DispatchFilter

### PR DESCRIPTION
**Pull request by davidyell**
References #51 

The aim of this pull request is to address the use of a Dispatch Filter, as they have been deprecated by the core since 3.3.0. [Book page for Dispatch filters](https://book.cakephp.org/3.0/en/development/dispatch-filters.html).

The book says to implement a middleware instead, however, there is no controller instance in the request to modify. Looking at the core plugin for Elastic Search, [they add directly to the Factory in the boostrap](https://github.com/cakephp/elastic-search/blob/master/config/bootstrap.php#L22-L23). 

I have use the same approach for this pull request, by deprecating the class and calling the Factory locator directly to add the Endpoint factory.

### CODE REVIEW

File: src/Routing/Filter/ControllerEndpointFilter.php [hunk 1]
```diff
@@ -9,6 +9,7 @@
  * This filter registers the endpoint model type in controllers
  *
  * @package Muffin\Webservice\Routing\Filter
+ * @deprecated Since 2.0.0
```
Comment (ADmad) [inline src/Routing/Filter/ControllerEndpointFilter.php]: The format for deprecated tag is `@deprecated <version> <description>`

File: config/bootstrap.php [hunk 2]
```diff
@@ -1,3 +1,2 @@
 <?php
-
-\Cake\Routing\DispatcherFactory::add('Muffin/Webservice.ControllerEndpoint');
+\Cake\Datasource\FactoryLocator::add('Endpoint', [Muffin\Webservice\Model\EndpointRegistry::class, 'get']);
```
Comment (ADmad) [inline config/bootstrap.php]: Do we have required integration tests to ensure this change works correctly?

Comment (davidyell) [reply config/bootstrap.php]: Whoops. That's me typing too fast, I've done that in #63 also.

File: src/Routing/Filter/ControllerEndpointFilter.php (hunk 1 above)
Comment (davidyell) [reply src/Routing/Filter/ControllerEndpointFilter.php]: Whoops. That's me typing too fast, I've done that in #63 also.

File: config/bootstrap.php (hunk 2 above)
Comment (ADmad) [reply config/bootstrap.php]: With `TableRegistry` being deprecated in core, we should ideally deprecate `EndpointRegistry` too and add and end point locator class similar to table locator :slightly_smiling_face:

Comment (davidyell) [reply config/bootstrap.php]: Good idea, I'll add an item to #51

**Comment by davidyell** [comment]
Please ignore the last few commits, I pushed accidentally, and I will update with a force push later.

File: tests/TestCase/BootstrapTest.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        $controller = new Controller();
+        $endpoint = $controller->loadModel('Test', 'Endpoint');
+
+        $this->assertInstanceOf(TestEndpoint::class, $endpoint);
+        $this->assertEquals('Test', $endpoint->getAlias());
+    }
+
+    public function testFactoryLocatorAddition()
+    {
+        $expected = [\Muffin\Webservice\Model\EndpointRegistry::class, 'get'];
```
Comment (davidyell) [inline tests/TestCase/BootstrapTest.php]: It's just dawned on me reviewing this, that this test case relies on the previous one. Should the call to `Plugin::load` be moved to `setUp()` ?

**Comment by davidyell** [comment]
It seems Travis is upset by the tests with `Using Query::repository() as getter is deprecated. Use getRepository() instead.`

I'll add a task to #51 for that.

**Comment by davidyell** [comment]
Hey @ADmad anything outstanding on this preventing it from being merged?

**Pull request merged**